### PR TITLE
base64 extension v1

### PR DESCRIPTION
**Pull request by DanielSinclair**
## Description

quick commands to encode or decode the clipboard contents from/to base64

## Type of change

- New extension

## Screencast

[Quick snippet demo clip](https://streamable.com/x4dw6t)

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I checked that extension's icon has size of `512x512`

### CODE REVIEW

File: extensions/base64/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "base64",
+  "title": "Base64",
+  "description": "quickly encode or decode base64 from the clipboard",
```
Comment (pernielsentikaer) [inline extensions/base64/package.json:5]:
```suggestion
  "description": "Quickly encode or decode base64 from the clipboard",
```

File: extensions/base64/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  "icon": "command-icon.png",
+  "author": "DanielSinclair",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "encode",
+      "title": "Base64 Encode",
+      "description": "overwrite the clipboard contents with an encoded string",
```
Comment (pernielsentikaer) [inline extensions/base64/package.json:13]:
```suggestion
      "description": "Overwrite the clipboard contents with an encoded string",
```

File: extensions/base64/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      "mode": "no-view"
+    },
+    {
+      "name": "decode",
+      "title": "Base64 Decode",
+      "description": "overwrite the clipboard contents with a decoded string",
```
Comment (pernielsentikaer) [inline extensions/base64/package.json:19]:
```suggestion
      "description": "Overwrite the clipboard contents with a decoded string",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @DanielSinclair 👋 

Thanks for your PR, I have a few NITs as suggested and a comment:

- If the clipboard is empty, then I propose that it show a toast(showToast) with an error explaining that nothing is in the clipboard to help with the UX

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by pernielsentikaer** [comment]
Hi @DanielSinclair 👋 

Do you need any help?

**Comment by DanielSinclair** [comment]
Hey @pernielsentikaer. Implemented those UX changes this evening!

File: extensions/base64/src/util/clipboard.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { runAppleScript } from "run-applescript"
+import { copyTextToClipboard, showHUD } from "@raycast/api"
+
+const isEmpty = (string: string) => {
+  return (string.length === 0)
+}
+
+export const contents = async () => {
+  const clipboard = await runAppleScript('the clipboard')
+  if (isEmpty(clipboard)) throw "clipboard is empty"
```
Comment (pernielsentikaer) [inline extensions/base64/src/util/clipboard.ts:10]:
```suggestion
  if (isEmpty(clipboard)) throw "Clipboard is empty"
```
Just a nitpick 😄

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @DanielSinclair 

Much better 💪 Really doing something good for the UX here! 💯 

IEkgdGhpbmsgaXQncyBhIGhhbmR5IHRvb2wgdGhhdCBjb3VsZCBiZWNvbWUgdmVyeSBoYW5keSBmb3IgbWFueSwgZ29vZCBqb2Ih 👏 

Just the last NIT and we're ready to ship it, I'm going to accept it later today regardless, but I give you a few hours to address it 😄

**Review by mattisssa — APPROVED** [review]

**Pull request merged**